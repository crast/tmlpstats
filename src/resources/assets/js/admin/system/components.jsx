import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'

import Api from '../../api'
import {
    Form, formActions, SimpleField, SimpleFormGroup,
    SimpleSelect, NullableTextControl, NullableTextAreaControl, BooleanSelect } from '../../reusable/form_utils'
import { ALERT_TYPES } from '../../reusable/ui_basic'
import { objectAssign } from '../../reusable/ponyfill'
import debounceRender from '../../reusable/debounce'
import { connectRedux, rebind, delayDispatch } from '../../reusable/dispatch'
import { loadStateShape } from '../../reusable/shapes'
import { SystemMessages } from '../../reusable/system-messages'

import { systemMessagesData } from '../../lookups/lookups-system-messages'
const MODEL = 'admin.system.currentMessage'

// Avoid a potentially laggy set of constant re-rendering markdown
const DebouncedMessages = debounceRender(SystemMessages, 300, {maxWait: 4000})

@connectRedux()
export class AdminSystemMessages extends React.Component {
    static mapStateToProps(state) {
        const { currentMessage } = state.admin.system
        const { loadState, saveState } = state.lookups
        return {
            systemMessages: systemMessagesData.selector(state),
            loadState, saveState, currentMessage
        }
    }

    static propTypes = {
        systemMessages: PropTypes.instanceOf(Immutable.Map),
        loadState: loadStateShape,
        saveState: loadStateShape,
        currentMessage: PropTypes.object
    }

    constructor(props) {
        super(props)
        rebind(this, 'onSubmit', 'onSelect')
    }

    checkLoading() {
        const { systemMessages, loadState } = this.props
        if (systemMessages.size) {
            return true
        }
        if (loadState.available) {
            delayDispatch(this, systemMessagesData.manager.runNetworkAction('load', {}, {
                setLoaded: true,
                api: Api.Admin.System.allSystemMessages,
                successHandler(data, { dispatch }) {
                    let m = Immutable.Map()
                    data.forEach((d) => {
                        m = m.set(d.id, d)
                    })
                    dispatch(systemMessagesData.replaceItems(m))
                }
            }))
        }
        return false
    }

    render() {
        const { systemMessages, loadState, currentMessage } = this.props
        if (!this.checkLoading()) {
            return <div>Loading messages</div>
        }

        let items = []
        systemMessages.forEach((message) => {
            let query = 'All Regions'
            if (message.region) {
                query = `Region ${message.region}`
            } else if (message.center) {
                query = `Center ${message.center}`
            }
            items.push(
                <tr key={message.id}>
                    <td><a href="#" onClick={this.onSelect} data-mid={message.id}>{message.id}</a></td>
                    <td>{message.active? 'Y' : 'N'}</td>
                    <td>{message.section}</td>
                    <td>{query}</td>
                    <td>{message.title}</td>
                </tr>
            )
        })

        return (
            <div>
                <table className="table">
                    <thead>
                        <tr>
                            <td>ID</td>
                            <td>Active</td>
                            <td>Section</td>
                            <td>Shown For</td>
                            <td>Title</td>
                        </tr>
                    </thead>
                    <tbody>
                        {items}
                    </tbody>
                </table>
                <h4>{currentMessage.id? `edit message ${currentMessage.id}` : 'new message'}</h4>
                <Form model={MODEL} onSubmit={this.onSubmit} className="form-horizontal">
                    <SimpleField model={MODEL+'.section'} label="Section" />
                    <SimpleField model={MODEL+'.region'} label="Region ID" />
                    <SimpleFormGroup label="Center ID" divClass="col-md-3">
                        <NullableTextControl model={MODEL+'.center'} />
                    </SimpleFormGroup>
                    <SimpleFormGroup label="Active" divClass="col-md-2">
                        <BooleanSelect model={MODEL+'.active'} />
                    </SimpleFormGroup>
                    <SimpleFormGroup label="Level" divClass="col-md-3">
                        <SimpleSelect model={MODEL+'.level'} items={ALERT_TYPES} />
                    </SimpleFormGroup>
                    <SimpleField model={MODEL+'.title'} label="Title" />
                    <SimpleFormGroup label="Content (markdown)">
                        <NullableTextAreaControl model={MODEL+'.content'} rows={10} />
                    </SimpleFormGroup>
                    <SimpleFormGroup label="Preview">
                        <DebouncedMessages messages={Immutable.List.of(currentMessage)} />
                    </SimpleFormGroup>
                    <button type="submit" className="btn btn-primary">Save</button>
                </Form>
            </div>
        )
    }

    onSelect(e) {
        const id = e.target.dataset.mid
        let message = this.props.systemMessages.get(parseInt(id))
        if (message) {
            this.props.dispatch(formActions.load(MODEL, message))
        }
    }

    onSubmit(currentMessage) {
        // clone so we can fix
        currentMessage = objectAssign({}, currentMessage)
        const centerRegion = ['center', 'region']
        centerRegion.forEach((k) => {
            currentMessage[k] = currentMessage[k] || null  // Override with null
        })

        this.props.dispatch(systemMessagesData.manager.runNetworkAction('save', {data: currentMessage}, {
            api: Api.Admin.System.writeSystemMessage,
            successHandler(data, { dispatch }) {
                currentMessage.id = data.storedId
                dispatch(systemMessagesData.replaceItem(currentMessage))
                dispatch(formActions.load(MODEL, {active: true}))
            }
        }))
    }
}
