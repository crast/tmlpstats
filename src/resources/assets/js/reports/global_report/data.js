import Immutable from 'immutable'

import { TabbedReportManager } from '../tabbed_report/manager'
import Api from '../../api'

const _baseKey = Immutable.Record({
    regionAbbr: '',
    reportingDate: '',
    page: null
})

export class GlobalReportKey extends _baseKey {
    queryParams() {
        return {region: this.regionAbbr, reportingDate: this.reportingDate}
    }
}



export const reportData = new TabbedReportManager({
    prefix: 'reports/GlobalReport',
    findRoot: (state) => state.reports.global_report,
    actions: {
        load: {
            api: Api.GlobalReport.getReportPagesByDate
        }
    }
})
