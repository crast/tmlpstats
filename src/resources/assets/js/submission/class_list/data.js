import SortableCollection, { compositeKey } from '../../reusable/sortable_collection'
import { LoadingMultiState, InlineBulkWork } from '../../reusable/reducers'

export const classListSorts = [
    {
        key: 'teamYear_first_last',
        label: 'Default',
        comparator: compositeKey([['teamYear', 'number'], ['firstName', 'string'], ['lastName', 'string']])
    },
    {
        key: 'first_last',
        label: 'First, Last',
        comparator: compositeKey([['firstName', 'string'], ['lastName', 'string']])
    },
    {
        key: 'last_first',
        label: 'Last, First',
        comparator: compositeKey([['lastName', 'string'], ['firstName', 'string']])
    }
]

export const teamMembersCollection = new SortableCollection({
    name: 'submission.class_list',
    key_prop: 'id',
    sort_by: 'teamYear_first_last',
    sorts: classListSorts
})

export const classListLoad = new LoadingMultiState('class_list/initialLoadState')
export const weeklyReportingSave = new LoadingMultiState('class_list/saveWeeklyReporting')
export const weeklyReportingData = new InlineBulkWork('class_list/weeklyReporting')
