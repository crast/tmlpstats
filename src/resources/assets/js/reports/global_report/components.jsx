import React, { Component } from 'react'
import { createSelector } from 'reselect'
import ReportsMeta from '../meta'
import { TabbedReport } from '../tabbed_report/components'
import { filterReportFlags, makeTwoTabParamsSelector } from '../tabbed_report/util'
import { reportData } from './data'
import { connectRedux, delayDispatch } from '../../reusable/dispatch'

const DEFAULT_FLAGS = {afterClassroom2: false}

@connectRedux()
export class GlobalReport extends Component {
    static mapStateToProps(state) {
        return reportData.opts.findRoot(state)
    }

    constructor(props) {
        super(props)

        // XXX TODO deal with report flags actually changing after classroom 2
        this.fullReport = filterReportFlags(ReportsMeta['Global'], DEFAULT_FLAGS)
        this.makeTabParams = makeTwoTabParamsSelector()
        const { params } = props
        if (params) {
            this.showReport(params.tab2 || params.tab1)
        }
    }

    reportUriBase() {
        const { regionAbbr, reportingDate } = this.props.params
        return `/reports/regions/${regionAbbr}/${reportingDate}`
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.params !== this.props.params) {
            this.showReport(nextProps.params.tab2 || nextProps.params.tab1)
        }
    }

    reportUri(parts) {
        let tabParts = parts.join('/')
        return `${this.reportUriBase()}/${tabParts}`
    }

    showReport(reportId) {
        const report = this.fullReport[reportId]
        if (!report) {
            alert('Unknown report page: ' + reportId)
        } else if (report.type != 'grouping') {
            const { regionAbbr, reportingDate } = this.props.params
            delayDispatch(this, reportData.loadReport(reportId, {region: regionAbbr, reportingDate}))
        }
    }

    getContent(reportId) {
        return this.props.data.get(reportId) || ''
    }

    responsiveLabel(report) {
        return responsiveLabel(report)
    }

    render() {
        const tabs = this.makeTabParams(this.props.params)
        return <TabbedReport tabs={tabs} fullReport={this.fullReport} reportContext={this} />
    }

}

/** Generate tab label HTML for a report if shortName is set */
function responsiveLabel(report) {
    if (report.shortName) {
        return [
            <span className="long" key="long">{report.name}</span>,
            <span className="brief" key="brief">{report.shortName}</span>
        ]
    } else {
        return report.name
    }
}
