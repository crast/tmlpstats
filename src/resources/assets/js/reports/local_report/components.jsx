import React, { Component } from 'react'

import { connectRedux, delayDispatch } from '../../reusable/dispatch'
import { TabbedReport } from '../tabbed_report/components'
import { filterReportFlags, makeTwoTabParamsSelector } from '../tabbed_report/util'
import ReportsMeta from '../meta'

import * as actions from './actions'
import * as pages from './pages'
import { reportData, LocalKey } from './data'

@connectRedux()
export class LocalReport extends Component {
    static mapStateToProps() {
        return (state) => {
            const { local_report: local, reportConfig } = state.reports
            return {
                pageData: local.reportData.data,
                config: reportConfig
            }
        }
    }

    constructor(props) {
        super(props)
        this.makeTabParams = makeTwoTabParamsSelector()
        // componentWillReceiveProps is not called on construction, it's easier to run the logic one time
        this.componentWillReceiveProps(props)
    }

    /**
     * componentWillReceiveProps is called whenever props change, or similarly redux props.
     * In the case of this report, it does a few things:
     *   1. If the params change (URL changes) then bump the selected report to top of queue.
     *   2. Fetches the report config and filters the report based on flags.
     */
    componentWillReceiveProps(nextProps) {
        const { params: nextParams, dispatch } = nextProps
        if (!this.storageKey || nextParams !== this.props.params) {
            this.storageKey = new LocalKey(nextParams) // will only grab centerId and reportingDate
            this.showReport(nextParams.tab2 || nextParams.tab1)
        }
        const { loadState, data } = nextProps.config
        const config = data.get(this.storageKey)
        if (!config) {
            if (loadState.available) {
                dispatch(actions.loadConfig(this.storageKey))
            }
        } else if (config !== this.savedConfig) {
            const r = this.fullReport = filterReportFlags(ReportsMeta['Local'], config.flags)
            this.savedConfig = config
            dispatch(reportData.init(r))
        }
        return config
    }

    reportUriBase() {
        const { centerId, reportingDate } = this.props.params
        return `/reports/centers/${centerId}/${reportingDate}`
    }

    /**
     * Required by the TabbedReport interface; builds a URI for a given tab
     * @param  array   parts URL parts; ['tab1Name', 'tab2Name']
     * @return string  The URI
     */
    reportUri(parts) {
        let tabParts = parts.join('/')
        return `${this.reportUriBase()}/${tabParts}`
    }

    /**
     * Required by the TabbedReport interface; gets page content
     * @param  string  reportId  The report's identifier e.g. 'FooBar'
     * @return string|object     The data for the page content.
     */
    getContent(reportId) {
        return this.props.pageData[reportId] || ''
    }

    // Dispatches an action on delay to prioritize/load a given report tab.
    showReport(reportId) {
        if (!this.fullReport) {
            setTimeout(() => this.showReport(reportId), 200)
            return
        }
        const report = this.fullReport[reportId]
        if (!report) {
            alert('Unknown report page: ' + reportId)
        } else if (report.type != 'grouping') {
            const params = this.storageKey.queryParams()
            delayDispatch(this, reportData.loadReport(reportId, params))
        }
    }

    responsiveLabel(report) {
        return responsiveLabel(report)
    }

    pageComponent(report) {
        return pages[report.id]
    }

    render() {
        if (!this.fullReport) {
            return <div>Loading...</div>
        }
        const tabs = this.makeTabParams(this.props.params)
        return <TabbedReport tabs={tabs} fullReport={this.fullReport} reportContext={this} />
    }
}

/** Generate tab label HTML for a report if shortName is set */
function responsiveLabel(report) {
    if (report.shortName) {
        return [
            <span className="long" key="long">{report.name}</span>,
            <span className="brief" key="brief">{report.shortName}</span>
        ]
    } else {
        return report.name
    }
}
