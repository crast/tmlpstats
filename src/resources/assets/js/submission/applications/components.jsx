import { Link, withRouter } from 'react-router'
import { connect } from 'react-redux'

import { Form, SimpleField, SimpleSelect, SimpleFormGroup, BooleanSelect, AddOneLink } from '../../reusable/form_utils'
import { Promise, objectAssign } from '../../reusable/ponyfill'
import { rebind } from '../../reusable/dispatch'
import { collectionSortSelector } from '../../reusable/sort-helpers'
import { ModeSelectButtons, LoadStateFlip, MessagesComponent, scrollIntoView } from '../../reusable/ui_basic'

import { SubmissionBase, React } from '../base_components'
import { APPLICATIONS_FORM_KEY } from './reducers'
import { appsSorts, appsCollection } from './data'
import { centerQuarterData } from '../core/data'
import { getLabelTeamMember } from '../core/selectors'
import { loadApplications, saveApplication, chooseApplication } from './actions'
import AppStatus from './AppStatus'

const getSortedApplications = collectionSortSelector(appsSorts)

class ApplicationsBase extends SubmissionBase {
    componentDidMount() {
        this.setupApplications()
    }

    setupApplications() {
        if (this.props.loading.state == 'new') {
            return this.props.dispatch(loadApplications(this.props.params.centerId, this.reportingDateString()))
        }
        return Promise.resolve(null)
    }

    getAppById(appId) {
        return this.props.applications.data[appId]
    }
}

class ApplicationsIndexView extends ApplicationsBase {
    constructor(props) {
        super(props)
        rebind(this, 'changeSort')
    }

    renderWithdrawsTable(apps) {
        if (!apps.length) {
            return <div />
        }

        const baseUri = this.baseUri()
        const withdraws = []

        apps.forEach((app) => {
            let key = app.id
            withdraws.push(
                <tr key={key}>
                    <td><Link to={`${baseUri}/applications/edit/${key}`}>{app.firstName} {app.lastName}</Link></td>
                    <td>{app.teamYear}</td>
                    <td>{app.regDate}</td>
                    <td>{app.wdDate}</td>
                    <td>{this.props.lookups.withdraw_codes_by_id[app.withdrawCode].display}</td>
                </tr>
            )
        })

        return (
            <div>
            <br/>
                <h4>Withdraws</h4>
                <table className="table">
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Year</th>
                            <th>Registered</th>
                            <th>Withdrawn</th>
                            <th>Reason</th>
                        </tr>
                    </thead>
                    <tbody>{withdraws}</tbody>
                </table>
            </div>
        )
    }

    render() {
        if (!this.props.loading.loaded) {
            return this.renderBasicLoading()
        }
        var apps = []
        var withdraws = []
        var baseUri = this.baseUri()
        const sortedApps = getSortedApplications(this.props.applications)

        sortedApps.forEach((app) => {
            const key = app.id
            const status = AppStatus.getStatusString(app)
            if (app.withdrawCode) {
                withdraws.push(app)
                return
            }

            apps.push(
                <tr key={key}>
                    <td><Link to={`${baseUri}/applications/edit/${key}`}>{app.firstName} {app.lastName}</Link></td>
                    <td>{app.teamYear}</td>
                    <td>{app.regDate}</td>
                    <td>{app.apprDate}</td>
                    <td>{status}</td>
                </tr>
            )
        })

        return (
            <div>
                <h3>Manage Registrations</h3>
                <ModeSelectButtons items={appsSorts} current={this.props.applications.meta.get('sort_by')}
                                   onClick={this.changeSort} ariaGroupDesc="Sort Preferences" />
                <table className="table">
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Year</th>
                            <th>Registered</th>
                            <th>Approved</th>
                            <th>Status</th>
                        </tr>
                    </thead>
                    <tbody>{apps}</tbody>
                </table>
                <AddOneLink link={`${baseUri}/applications/add`} />
                {this.renderWithdrawsTable(withdraws)}
            </div>
        )
    }

    changeSort(newSort) {
        this.props.dispatch(appsCollection.setMeta('sort_by', newSort))
    }
}

class _EditCreate extends ApplicationsBase {
    render() {
        const modelKey = APPLICATIONS_FORM_KEY
        const app = this.props.currentApp
        let messages = []
        if (app && app.id) {
            messages = this.props.messages[app.id]
        } else if (this.isNewApp() && this.props.messages['create']) {
            messages = this.props.messages['create']
        }

        return (
            <div>
                <h3>{this.title()}</h3>

                <MessagesComponent messages={messages} />

                <Form className="form-horizontal" model={modelKey} onSubmit={this.saveAppData.bind(this)}>
                    {this.renderStartingQuarter(modelKey)}
                    <SimpleField label="First Name" model={modelKey+'.firstName'} divClass="col-md-6" required={true} />
                    <SimpleField label="Last Name" model={modelKey+'.lastName'} divClass="col-md-6" required={true} />
                    <SimpleField label="Team Year" model={modelKey+'.teamYear'} divClass="col-md-1" required={true} customField={true}>
                        <select className="form-control">
                            <option value="1">Team 1</option>
                            <option value="2">Team 2</option>
                        </select>
                    </SimpleField>
                    <SimpleField label="Email" model={modelKey+'.email'} divClass="col-md-6" />
                    <SimpleField label="Comment" model={modelKey+'.comment'} divClass="col-md-6" customField={true}>
                        <textarea className="form-control" rows="3"></textarea>
                    </SimpleField>
                    <div className="required form-group">
                        <label className="col-md-2 control-label">Committed Team Member</label>
                        <div className="col-md-6">
                            <SimpleSelect
                                    model={modelKey+'.committedTeamMember'} items={this.props.lookups.team_members}
                                    keyProp="teamMemberId" getLabel={getLabelTeamMember} emptyChoice="Choose One" />
                        </div>
                    </div>
                    <div className="form-group">
                        <label className="col-md-2 control-label">Application Status</label>
                        <div className="col-md-9">
                            <AppStatus model={modelKey} currentApp={this.props.currentApp} lookups={this.props.lookups} dispatch={this.props.dispatch} />
                        </div>
                    </div>
                    {this.renderTravelRoom(modelKey)}
                    <div className="form-group">
                        <div className="col-sm-offset-2 col-sm-8">
                            <LoadStateFlip loadState={this.props.saveApp}>
                                <button className="btn btn-primary" type="submit">Save</button>
                            </LoadStateFlip>
                        </div>
                    </div>
                </Form>
            </div>
        )
    }

    // Return true if this is a new app
    isNewApp() {
        const { currentApp } = this.props
        return (!currentApp.id || parseInt(currentApp.id) < 0)
    }

    getCenterQuarters() {
        const cqData = this.props.centerQuarters.data
        const centerQuarters = this.props.lookups.validRegQuarters.map(id => cqData[id])

        return { cqData, centerQuarters }
    }

    renderStartingQuarter(modelKey) {
        const { currentApp } = this.props
        const { cqData, centerQuarters } = this.getCenterQuarters()
        let body = ''
        if (this.isNewApp()) {
            body = <SimpleSelect model={modelKey+'.incomingQuarter'} items={centerQuarters}
                                 keyProp="quarterId" getLabel={centerQuarterData.getLabel} />
        } else {
            const cq = cqData[currentApp.incomingQuarter]
            body = (cq)? centerQuarterData.getLabel(cq) : 'Unknown'
        }

        let requiredClass = ''
        if (this.isNewApp()) {
            requiredClass = 'required'
        }

        return (
            <div className={requiredClass + ' form-group'}>
                <label className="col-md-2 control-label">Starting Quarter</label>
                <div className="col-md-6">{body}</div>
            </div>
        )
    }

    renderTravelRoom(modelKey) {
        return (
            <div>
                <SimpleFormGroup label="Travel Booked" divClass="col-md-6 boolSelect">
                    <BooleanSelect model={modelKey+'.travel'} style={{maxWidth: '4em'}} />
                </SimpleFormGroup>
                <SimpleFormGroup label="Room Booked" divClass="col-md-6 boolSelect">
                    <BooleanSelect model={modelKey+'.room'} />
                </SimpleFormGroup>
            </div>
        )
    }

    // saveAppData for now is the same between edit and create flows
    saveAppData(data) {
        this.props.dispatch(saveApplication(this.props.params.centerId, this.reportingDateString(), data)).then((result) => {
            if (!result) {
                return
            }

            if (result.messages && result.messages.length) {
                scrollIntoView('react-routed-flow')

                // Redirect to edit view if there are warning messages
                if (this.isNewApp() && result.valid) {
                    this.props.router.push(this.baseUri() + '/applications/edit/' + result.storedId)
                }
            } else if (result.valid) {
                this.props.router.push(this.baseUri() + '/applications')
            }
        })
    }
}

class ApplicationsEditView extends _EditCreate {
    componentDidMount() {
        super.setupApplications().then(() => {
            var appId = this.props.params.appId
            if (!this.props.currentApp || this.props.currentApp.id != appId) {
                let app = this.getAppById(appId)
                if (app){
                    app = objectAssign({}, app, {committedTeamMember: app.committedTeamMember || ''})
                    this.props.dispatch(chooseApplication(appId, app))
                }
            }
        })
    }

    title() {
        return 'Edit Application'
    }

    render() {
        const appId = this.props.params.appId

        if (!this.props.loading.loaded
            || !this.props.currentApp
            || this.props.currentApp.id != appId
            || !this.props.lookups
            || !this.props.lookups.validRegQuarters
            || !this.props.lookups.team_members
        ) {
            return <div>{this.props.loading.state}...</div>
        }
        return super.render()
    }
}

class ApplicationsAddView extends _EditCreate {
    componentDidMount() {
        super.setupApplications().then(() => {
            if (this.props.currentApp) {
                const { centerQuarters } = this.getCenterQuarters()
                const blankApp = {
                    firstName: '',
                    lastName: '',
                    teamYear: 1,
                    regDate: this.reportingDateString(),
                    incomingQuarter: centerQuarters[0].quarterId,
                    committedTeamMember: '',
                }
                this.props.dispatch(chooseApplication('', blankApp))
            }
        })
    }

    title() {
        return 'Create Application'
    }

    render() {
        if (!this.props.loading.loaded
            || !this.props.lookups
            || !this.props.lookups.validRegQuarters
            || !this.props.lookups.team_members
        ) {
            return <div>{this.props.loading.state}...</div>
        }
        return super.render()
    }

}

const mapStateToProps = (state) => {
    const s = state.submission
    return objectAssign({lookups: s.core.lookups, centerQuarters: s.core.centerQuarters}, s.applications)
}
const connector = connect(mapStateToProps)

export const ApplicationsIndex = connector(ApplicationsIndexView)
export const ApplicationsEdit = connector(withRouter(ApplicationsEditView))
export const ApplicationsAdd = connector(withRouter(ApplicationsAddView))
