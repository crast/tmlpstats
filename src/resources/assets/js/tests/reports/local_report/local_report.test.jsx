import React from 'react'
import renderer from 'react-test-renderer'
import Immutable from 'immutable'
import moment from 'moment'

import { Wrap, store } from '../../testing-store'

import { LocalReport } from '../../../reports/local_report/components'
import { reportConfigData } from '../../../reports/data'
import { LocalKey, reportData } from '../../../reports/local_report/data'
import { lookupsData } from '../../../lookups'


const regionCentersData = lookupsData.scopes.region_centers


describe('LocalReport', () => {
    const baseParams = {centerId: 'abc', reportingDate: '2017-04-07'}
    const key = new LocalKey(baseParams)
    let reportConfig = {
        'statsReportId':12345,
        'globalRegionId':'na',
        'flags':{'canReadContactInfo':true, 'firstWeek': true, 'nextQtrAccountabilities': false},
        'capabilities':{'reportToken':true,'reportNavLinks':true},
        'centerInfo':{'id':16,'name':'ABC','abbreviation':'ABC','teamName':null,'regionId':2},
        'reportToken':'http:\/\/localhost\/report\/abc'
    }
    store.dispatch(regionCentersData.replaceItem('na', Immutable.OrderedMap([
        [1, {id: 1, name: 'AAA', 'abbreviation': 'aaa'}],
        [2, {id: 2, name: 'ABC', 'abbreviation': 'abc'}],
        [3, {id: 3, name: 'DEF', 'abbreviation': 'def'}]
    ])))

    test('Basic', () => {
        store.dispatch(reportConfigData.replaceItem(key, reportConfig))
        store.dispatch(reportData.replaceItem(key.set('page', 'Summary'), 'Summary Page Here'))
        const params = Object.assign({tab1: 'Summary'}, baseParams)
        const tree = renderer.create(
            <Wrap>
                <LocalReport params={params} />
            </Wrap>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Next Qtr Accountabilities', () => {
        reportConfig.flags.nextQtrAccountabilities = true
        store.dispatch(reportConfigData.replaceItem(key, reportConfig))
        store.dispatch(reportData.replaceItem(key.set('page', 'NextQtrAccountabilities'), {nqas: [
                {id: 123, accountability: {display: 'Boss'}, name: 'Person Name', phone: '12345', email: 'abc@def.com', meta: {updatedAt: moment('2017-01-01T02:03:04').utc().format()}}
        ]}))
        const params = Object.assign({tab1: 'NextQtrAccountabilities'}, baseParams)
        const tree = renderer.create(
            <Wrap>
                <LocalReport params={params} />
            </Wrap>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
