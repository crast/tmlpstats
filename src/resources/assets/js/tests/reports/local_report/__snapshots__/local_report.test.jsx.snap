// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalReport Basic 1`] = `
<div>
  <h2>
    ABC
     - 
    2017-04-07
     
    <button
      className="btn btn-default"
      onClick={[Function]}
    >
      Show Report Link
    </button>
  </h2>
  <div
    className="row"
  >
    <div
      className="col-sm-8"
    >
      <a
        href="/home"
      >
        « See All
      </a>
    </div>
    <div
      className="col-sm-4"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <a
        className="btn btn-default"
        onClick={[Function]}
        style={Object {}}
      >
        « 
        AAA
      </a>
      <a
        className="btn btn-default"
        onClick={[Function]}
        style={Object {}}
      >
        DEF
         »
      </a>
    </div>
  </div>
  <div>
    <ul
      className="nav nav-tabs tabs-top brief-tabs"
      id="tabs"
    >
      <li
        className="active"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Weekly Summary
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Report Details
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Center Games
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Team Members
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Team Expansion
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Courses
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="long"
          >
            Contact Info
          </span>
          <span
            className="brief"
          >
            Contacts
          </span>
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Transfer Check
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="long"
          >
            Weekend Information
          </span>
          <span
            className="brief"
          >
            Weekend
          </span>
        </a>
      </li>
    </ul>
    <div>
      <div
        className="tab-content"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Summary Page Here",
            }
          }
          id="content-Summary"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`LocalReport Next Qtr Accountabilities 1`] = `
<div>
  <h2>
    ABC
     - 
    2017-04-07
     
    <button
      className="btn btn-default"
      onClick={[Function]}
    >
      Show Report Link
    </button>
  </h2>
  <div
    className="row"
  >
    <div
      className="col-sm-8"
    >
      <a
        href="/home"
      >
        « See All
      </a>
    </div>
    <div
      className="col-sm-4"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <a
        className="btn btn-default"
        onClick={[Function]}
        style={Object {}}
      >
        « 
        AAA
      </a>
      <a
        className="btn btn-default"
        onClick={[Function]}
        style={Object {}}
      >
        DEF
         »
      </a>
    </div>
  </div>
  <div>
    <ul
      className="nav nav-tabs tabs-top brief-tabs"
      id="tabs"
    >
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Weekly Summary
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Report Details
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Center Games
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Team Members
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Team Expansion
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Courses
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="long"
          >
            Contact Info
          </span>
          <span
            className="brief"
          >
            Contacts
          </span>
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          Transfer Check
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="long"
          >
            Weekend Information
          </span>
          <span
            className="brief"
          >
            Weekend
          </span>
        </a>
      </li>
      <li
        className="active"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="long"
          >
            Next Quarter Accountabilities
          </span>
          <span
            className="brief"
          >
            Accountabilities
          </span>
        </a>
      </li>
    </ul>
    <div>
      <div
        className="tab-content"
      >
        <div
          className="table-responsive"
        >
          <h3>
            Next Quarter Accountabilities
          </h3>
          <table
            className="table table-condensed table-striped table-hover"
            id="tbl-localReport_nqa"
          >
            <thead>
              <tr>
                <th
                  aria-controls="tbl-localReport_nqa"
                  className="tabular-sortable"
                  onClick={[Function]}
                >
                  Accountability
                </th>
                <th
                  aria-controls="tbl-localReport_nqa"
                  className="tabular-sortable"
                  onClick={[Function]}
                >
                  Name
                </th>
                <th
                  aria-controls="tbl-localReport_nqa"
                  className="tabular-sortable"
                  onClick={[Function]}
                >
                  Phone
                </th>
                <th
                  aria-controls="tbl-localReport_nqa"
                  className="tabular-sortable"
                  onClick={[Function]}
                >
                  Email
                </th>
                <th
                  aria-controls="tbl-localReport_nqa"
                  className="tabular-sortable"
                  onClick={[Function]}
                >
                  Updated At
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  Boss
                </td>
                <td>
                  Person Name
                </td>
                <td>
                  12345
                </td>
                <td>
                  abc@def.com
                </td>
                <td>
                  Jan 1, 2017 02:03 AM
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
